### PR TITLE
FEATURE: Add edit page for personal schedules

### LINKED ISSUE #5: FEATURE: Add edit page for personal schedules
### Have you updated the title?

- [X] I have updated the issue's title.

### As a...

user

### I can...

click "Edit" in the personal schedule table

### So that...

I can edit my personal schedule

### Discussion

Currently the personal schedule edit page is a placeholder, replace this with a form or table to allow students to delete courses from their personal schedule.

### Acceptance Criteria

- [x] There is a form or table which displays a specific personal schedule & allows the user to delete courses.
- [x] This form or table has been added to the personal schedule edit page.
- [x] Deleting courses from a personal schedule deletes them from the PSCourse repository.

### Implementation Todos

Frontend:
- [x] Add a Personal Schedule Edit Table (or similar), which allows the user to view the courses in a schedule & delete any courses.
- [x] This edit table should make API calls to the PSCourse API to delete courses 
- [x] Add this edit table to the Personal Schedule Edit Page

Testing:
- [x] Front end tests pass and there is adequate coverage
- [x] Back end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by theanimated01**
Closes #5 

Dev deployment: https://proj-courses-theanimated01-dev.dokku-07.cs.ucsb.edu

This PR adds the edit page feature for personal schedules where you can delete courses from the schedule page. Deleting a course from this page also deletes it from PSCourse Repository.

Changes made:

Backend:
1. Added a new delete mapping to delete a course by enroll code (enrollCd) and personal schedule id (psId). 
2. Also added tests for the new delete mapping

Frontend:
1. Created a new table called PersonalSectionsEditTable which has the delete button and functionality to delete a course
2. Added tests for this new table
3. Created a new utils file called CourseEditUtlis which is used to call the new delete mapping from the frontend
4. Added tests for this new utils file
5. Modified the placeholder PersonalSchedulesEditPage to contain the PersonalScheduleTable and the PersonalSectionsEditTable
6. Added tests for the modified edit page

Also ran npx prettier --write "**/*.js" which fixed formatting issues for multiple existing files

Screenshots for before and after:
EditPage_before
EditPage_after

### CODE REVIEW

**Review by bwr02 — APPROVED** [review]
LGTM
but it also seems like your implementation is more than 10 points 😭

File: .gitignore
Comment (sophiattran) [inline .gitignore:1]: You might not want to upload this file

**Review by sophiattran — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (theanimated01) [reply .gitignore:1]: i did that on purpose. i edited it to include .DS_store files that mac makes so that it's not accidentally pushed to the repo

Comment (sophiattran) [reply .gitignore:1]: lgtm then!

**Review by sophiattran — APPROVED** [review]

**Review by tallyhawley — APPROVED** [review]
Looks good to me. Once this is merged, please stress to everyone that ALL active dev branches should be rebased on top of this commit due to the `prettier` changes, to resolve possible merge conflicts early on.

**Review by Shakcmpsc — APPROVED** [review]

**Pull request merged**